doc/source/data/key-concepts.rst

```diff
@@ -146,6 +146,16 @@ scheduling strategy for all Datasets tasks/actors, using the global
   :start-after: __resource_allocation_begin__
   :end-before: __resource_allocation_end__
 
+Fault tolerance
+===============
+
+Datasets relies on :ref:`task-based fault tolerance <task-fault-tolerance>` in Ray core. Specifically, if the driver that created a ``Dataset`` is still alive, then Ray core will guarantee that the underlying data is still accessible. This works through **lineage reconstruction**: a Dataset is a collection of Ray objects stored in shared memory, and if any of these objects are lost, then Ray will recreate these by re-executing the task(s) that created them.
```

Comment:
Is it possible that those objects (of the same dataset) owned by different processes?

Reply:
No, objects in the same Dataset should always be owned by the caller.